torch/export/_trace.py

```diff
@@ -1873,7 +1874,7 @@ def _export_for_training(
     kwargs: Optional[Dict[str, Any]] = None,
     dynamic_shapes: Optional[Union[Dict[str, Any], Tuple[Any], List[Any]]] = None,
     *,
-    strict: bool = True,
+    strict: bool = False if sys.platform == "win32" else True,
```

Comment:
In general I think this is confusing to people as our behavior is not consistent across platforms. On the other hand if there's <1% of people actually using windows, I'm kinda fine with this change.